krux_boto/__init__.py

```diff
@@ -33,6 +34,20 @@
 NAME            = 'krux-boto'
 DEFAULT_REGION  = 'us-east-1'
 
+
+def get_instance_region():
+    """
+    Query the instance metadata service and return the region this instance is
+    placed in. If the metadata service can't be contacted, return a generic
+    default instead.
+    """
+    ### TODO: XXX This shouldn't get called if we're not on EC2.
+    zone = boto.utils.get_instance_metadata()['placement']['availability-zone']
```

Comment:
you probably want to use a 'get' here to avoid any KeyErrors having to be trapped in case either the format changes or the region can't be contacted.

Also, add at least a debug statement if the default is being hit